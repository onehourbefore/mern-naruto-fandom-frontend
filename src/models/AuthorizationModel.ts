import { StatusModel } from "./StatusModel"

export type UserDataType = {
    id: string,
    email: string,
    name: string,
    role: string,
    avatar: string,
    created: number,
    liked: string [] | []
}

export type ResponseType = {
    accessToken: string,
    refreshToken: string,
    user: UserDataType
}

export interface AuthorizationState {
    user: UserDataType,
    status: StatusModel,
    deleteProfileStatus: StatusModel
}