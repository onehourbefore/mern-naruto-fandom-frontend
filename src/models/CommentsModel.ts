import { StatusModel } from "./StatusModel"

export type CommentType = {
    _id: string,
    postID: string,
    date: string,
    name: string,
    email: string,
    body: string,
    avatar: string,
}

export interface CommentsState {
    comments: [] | CommentType [],
    status: StatusModel,
}

export type CommentProps = {
    data: {
        _id: string,
        name: string,
        email: string,
        body: string,
        avatar: string,
        date: string,
    },
    postID: string | undefined,
    commentIndex: number,
}

export type CommentFormProps = {
    setVisible: (arg: boolean) => void,
}

export type CommentFromSidebarProps = {
    comment: CommentType,
}