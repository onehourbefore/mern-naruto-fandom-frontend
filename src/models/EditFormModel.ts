

export enum TasksEnum {
    CREATE= 'create',
    UPDATE = 'update',
    IDLE = 'idle'
}

export interface EditFormState {
    postID: string
    task: TasksEnum
}