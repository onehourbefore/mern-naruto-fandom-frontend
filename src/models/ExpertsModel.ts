import { StatusModel } from "./StatusModel"

export type ExpertType = {
    avatar: string,
    email: string,
    name: string,
    created: number
}


export interface ExpertsState {
    experts: [] | ExpertType [],
    status: StatusModel
}