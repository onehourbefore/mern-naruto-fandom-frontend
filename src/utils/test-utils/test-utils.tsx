import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { setupStore } from './testStore';
import type { AppStore, RootState } from './testStore';


interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
};


export function renderWithProviders (
  ui: { elem: React.ReactElement, path?: string },
  // routeComponent?: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {} ) 
  
{
  function Wrapper ({ children }: PropsWithChildren <{}>): JSX.Element {
    return (
      <MemoryRouter initialEntries={[ui.path ?? '/']}>
        <Provider store={store}>
          {/* {routeComponent} */}
          {children}
        </Provider>
      </MemoryRouter>
    )
  }

  return { store, ...render (ui.elem, { wrapper: Wrapper, ...renderOptions }) }
};