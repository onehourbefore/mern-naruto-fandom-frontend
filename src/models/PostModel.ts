import { StatusModel } from "./StatusModel"
import { CommentType } from "./CommentsModel"

export type PostType = {
    _id: string,
    author: string,
    title: string,
    content: string,
    tags: string [],
    likes: number,
    comments: [] | CommentType [],
    views: number,
    image: string
}

export interface PostsState {
    all: { posts: PostType [] | [], status: StatusModel },
    created: { post: PostType | null, status: StatusModel },
    updated: { post: PostType | null, status: StatusModel },
    deleted: { post: PostType | null, status: StatusModel },
    count: { count: number, status: StatusModel }
}

export interface OnePostState {
    active: { post: PostType | null, status: StatusModel },
    likeStatus: StatusModel,
    commentStatus: StatusModel
}

export type FeedbackPanelProps = {
    activePost: PostType,
    id: string | undefined
}

export type PostProps = {
    post: PostType
}

export type PostInfoProps = {
    data: {
        likes: number,
        comments: [] | CommentType [],
        views: number
    }
}

export type PostContentProps = {
    data: {
        author: string,
        title: string,
        content: string,
        tags: string []
    }
}

export type PostDataProps = {
    activePost: PostType
}

export type PostTypeForCreate = {
    title: string,
    content: string,
    tags: string,
    image: Blob | string,
}


