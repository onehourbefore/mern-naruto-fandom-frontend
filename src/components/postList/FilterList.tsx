import React from 'react';

import { PostType } from '../../models/PostModel';
import Post from '../post/Post';

import cl from './PostList.module.scss';


type FilterListProps = {
    posts: PostType [],
}

const FilterList: React.FC <FilterListProps> = (
    { posts }
): JSX.Element => {
    return (
        <div
            data-testid='filterlist-test'
            className={cl.root}
        >
            {posts.map ((post) => 
                <Post key={post._id} post={post} />)}
        </div>
    )
}

export default FilterList