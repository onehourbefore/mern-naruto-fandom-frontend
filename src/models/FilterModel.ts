import { StatusModel } from "./StatusModel"
import { PostType } from "./PostModel"

export interface FilterState {
    list: [] | PostType [],
    tags: [] | string [],
    fetchStatus: StatusModel,
    searchStatus: StatusModel
};