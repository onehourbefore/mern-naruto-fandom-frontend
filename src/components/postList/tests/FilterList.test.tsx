import { render, screen } from '@testing-library/react';
import { renderWithProviders } from '../../../utils/test-utils/test-utils';
import { PostType } from '../../../models/PostModel';
import FilterList from '../FilterList';

const posts: PostType[] = [
    {
        _id: 'zzzxxxccc1212',
        author: 'Jake',
        title: 'NARUTO',
        content: `Lorem ipsum dolor sit amet consectetur adipisicing elit. 
        Eos fuga deserunt, delectus alias quasi sint dolor, expedita 
        corporis aspernatur magni.`,
        tags: ['one', 'two'],
        likes: 21,
        comments: [],
        views: 213,
        image: 'narutoImage.jpg',
    },
    {
        _id: 'aaasss2211',
        author: 'Rob',
        title: 'SASUKE',
        content: `Voluptate numquam reprehenderit 
        molestias doloribus necessitatibus. Debitis atque hic facere.`,
        tags: ['three', 'four'],
        likes: 12,
        comments: [],
        views: 129,
        image: 'sasukeImage.jpg',
    },
];


describe('FilterList', () => {
    it('render with posts', () => {
        const component = renderWithProviders({ elem: <FilterList posts={posts}/> });
        expect(screen.getByText(/naruto/i)).toBeInTheDocument();
        expect(screen.getByText(/Lorem ipsum dolor sit amet/i)).toBeInTheDocument();
        expect(screen.getByText(/naruto/i)).toBeInTheDocument();
        expect(screen.getByText(/Voluptate numquam reprehenderit/i)).toBeInTheDocument();
        expect(component).toMatchSnapshot();
    });
});