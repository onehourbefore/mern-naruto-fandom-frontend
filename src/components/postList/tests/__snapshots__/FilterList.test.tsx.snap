// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList render with posts 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
        data-testid="filterlist-test"
      >
        <div
          class="root"
          data-testid="post-root"
        >
          <div
            class="root__mainWrapper"
          >
            <div
              class="root"
            >
              <div
                class="root__header"
              >
                <div
                  class="root__author"
                >
                  <span
                    class="root__author"
                  >
                    one
                     
                  </span>
                  <span
                    class="root__author"
                  >
                    two
                     
                  </span>
                </div>
                <div
                  class="root__author"
                >
                  Автор: 
                  Jake
                </div>
              </div>
              <div
                class="root__titleWrapper"
              >
                <h2
                  class="root__title"
                >
                  NARUTO
                </h2>
              </div>
              <div
                class="root__content"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. 
        Eos fuga deserunt, delectus alias quasi sint dolor, expedita 
        corporis aspernatur magni.
                ...
              </div>
            </div>
          </div>
        </div>
        <div
          class="root"
          data-testid="post-root"
        >
          <div
            class="root__mainWrapper"
          >
            <div
              class="root"
            >
              <div
                class="root__header"
              >
                <div
                  class="root__author"
                >
                  <span
                    class="root__author"
                  >
                    three
                     
                  </span>
                  <span
                    class="root__author"
                  >
                    four
                     
                  </span>
                </div>
                <div
                  class="root__author"
                >
                  Автор: 
                  Rob
                </div>
              </div>
              <div
                class="root__titleWrapper"
              >
                <h2
                  class="root__title"
                >
                  SASUKE
                </h2>
              </div>
              <div
                class="root__content"
              >
                Voluptate numquam reprehenderit 
        molestias doloribus necessitatibus. Debitis atque hic facere.
                ...
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
      data-testid="filterlist-test"
    >
      <div
        class="root"
        data-testid="post-root"
      >
        <div
          class="root__mainWrapper"
        >
          <div
            class="root"
          >
            <div
              class="root__header"
            >
              <div
                class="root__author"
              >
                <span
                  class="root__author"
                >
                  one
                   
                </span>
                <span
                  class="root__author"
                >
                  two
                   
                </span>
              </div>
              <div
                class="root__author"
              >
                Автор: 
                Jake
              </div>
            </div>
            <div
              class="root__titleWrapper"
            >
              <h2
                class="root__title"
              >
                NARUTO
              </h2>
            </div>
            <div
              class="root__content"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. 
        Eos fuga deserunt, delectus alias quasi sint dolor, expedita 
        corporis aspernatur magni.
              ...
            </div>
          </div>
        </div>
      </div>
      <div
        class="root"
        data-testid="post-root"
      >
        <div
          class="root__mainWrapper"
        >
          <div
            class="root"
          >
            <div
              class="root__header"
            >
              <div
                class="root__author"
              >
                <span
                  class="root__author"
                >
                  three
                   
                </span>
                <span
                  class="root__author"
                >
                  four
                   
                </span>
              </div>
              <div
                class="root__author"
              >
                Автор: 
                Rob
              </div>
            </div>
            <div
              class="root__titleWrapper"
            >
              <h2
                class="root__title"
              >
                SASUKE
              </h2>
            </div>
            <div
              class="root__content"
            >
              Voluptate numquam reprehenderit 
        molestias doloribus necessitatibus. Debitis atque hic facere.
              ...
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
